libpod/runtime_pod_common.go

```diff
@@ -162,13 +161,13 @@ func (r *Runtime) removeMalformedPod(ctx context.Context, p *Pod, ctrs []*Contai
 
 	// So, technically, no containers have been *removed*.
 	// They're still in the DB.
-	// So just return nil for removed containers. Squash all the errors into
-	// a multierror so we don't lose them.
+	// So just return nil for removed containers. Squash all the errors
+	// with errors.Join so we don't lose them.
 	if errored {
 		var allErrors error
 		for ctr, err := range removedCtrs {
 			if err != nil {
-				allErrors = multierror.Append(allErrors, fmt.Errorf("removing container %s: %w", ctr, err))
+				allErrors = errors.Join(allErrors, fmt.Errorf("removing container %s", ctr), err)
```

Comment:
this is not equivalent, you are now returning two error with no relationship whatsoever. The removing container must stay as part of the same error message otherwise you loose context